icu4c/source/i18n/dtitvfmt.cpp

```diff
@@ -540,6 +552,9 @@ DateIntervalFormat::setDateIntervalInfo(const DateIntervalInfo& newItvPattern,
                                         UErrorCode& status) {
     delete fInfo;
     fInfo = new DateIntervalInfo(newItvPattern);
+    if (fInfo == NULL) {
```

Comment:
The file is somewhat inconsistent in terms of using `NULL` or `nullptr`.
I tried to use the existing convention for each function in order to minimize churn.
However, I could just blanket change to use `nullptr` if that would be more clear.